CMakeLists.txt

```diff
@@ -38,56 +38,129 @@ set(SVT_HEVC_SOVERSION 1)
 
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_BUILD_TYPE}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_BUILD_TYPE}/lib)
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_BUILD_TYPE}/bin)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/Bin/${CMAKE_BUILD_TYPE}/)
+
+if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
+    set(CMAKE_INSTALL_LIBDIR "lib")
+endif()
+
+if(NOT DEFINED CMAKE_INSTALL_INCLUDEDIR)
+    set(CMAKE_INSTALL_INCLUDEDIR "include")
+endif()
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_C_STANDARD 99)
 set(CAN_USE_ASSEMBLER TRUE)
-set(CMAKE_NASM_LINK_EXECUTABLE "ld -o <TARGET> <OBJECTS>")
-set(CMAKE_ASM_NASM_SOURCE_FILE_EXTENSIONS asm)
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    set(CMAKE_ASM_NASM_OBJECT_FORMAT macho64)
-    set(CMAKE_ASM_NASM_FLAGS "-DUNIX64")
-    set(CMAKE_MACOSX_RPATH 1)
+if(WIN32)
+    set(CMAKE_ASM_NASM_FLAGS "${CMAKE_ASM_NASM_FLAGS} -DWIN64")
+else()
+    set(CMAKE_ASM_NASM_FLAGS "${CMAKE_ASM_NASM_FLAGS} -DUNIX64")
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    set(CMAKE_ASM_NASM_FLAGS "-DUNIX64")
-    set(CMAKE_ASM_NASM_FLAGS_DEBUG "-g null")
-    set(CMAKE_ASM_NASM_OBJECT_FORMAT elf64)
-    set(CMAKE_EXE_LINKER_FLAGS "-z noexecstack -z relro -z now -pie ")
-    set(CMAKE_SHARED_LINKER_FLAGS "-z noexecstack -z relro -z now -pie ")
+if(UNIX)
+    if(APPLE)
+        set(CMAKE_MACOSX_RPATH 1)
+        set(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
+        set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
+        set(CMAKE_C_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
+        set(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
+    else()
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -z noexecstack -z relro -z now -pie ")
+    endif()
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-    set(CMAKE_ASM_NASM_FLAGS "-DWIN64")
-    set(CMAKE_ASM_NASM_FLAGS_DEBUG "")
-    if (MSVC)
-        set(CMAKE_C_FLAGS " /GS /sdl ")
-        set(CMAKE_CXX_FLAGS " /GS /sdl ")
-        set(CMAKE_EXE_LINKER_FLAGS " /NXCompat /DynamicBase ")
-        set(CMAKE_SHARED_LINKER_FLAGS " /NXCompat /DynamicBase ")
+set(flags_to_test
+    -Wextra
+    -Wformat
+    -Wformat-security
+    -mavx
```

Comment:
should it be avx2 here? why don't use native anymore?

Reply:
I think it's been mavx, although I would prefer to not use mavx in top level since
`GCC depresses SSEx instructions when -mavx is used. Instead, it generates new AVX instructions or AVX equivalence for all SSEx instructions when needed.` from [here](https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html) and imo it's better to not use native unless the end user sets that as cflags because if they wish to compile once and copy to other servers, it might not work due to native flag

Reply:
Actually, thats a copy over from svt-av1, need to delete

Reply:
I've added an option to add back the march=native flag